### DISCUSSION TITLE
What's different between the null and empty string in validator

### DISCUSSION THREAD

**Discussion by cxlblm** [laravel/framework#33115] [General]
```php
$validator = Validator::make([
    'name' => 'jack',
    'age' => null,
], [
    'name' => 'required|string',
    'age' => 'integer',
]);

dump($validator->fails());  // true

$validator = Validator::make([
    'name' => 'jack',
    'age' => '',
], [
    'name' => 'required|string',
    'age' => 'integer',
]);

dump($validator->fails());  // false  why is passed?

**Comment by SudoGetBeer** [comment]
Hey,

Laravel checks every rule+value combination with this function:

```php
    /**
     * Determine if the attribute is validatable.
     *
     * @param  object|string  $rule
     * @param  string  $attribute
     * @param  mixed  $value
     * @return bool
     */
    protected function isValidatable($rule, $attribute, $value)
    {
        if (in_array($rule, $this->excludeRules)) {
            return true;
        }

        return $this->presentOrRuleIsImplicit($rule, $attribute, $value) &&
               $this->passesOptionalCheck($attribute) &&
               $this->isNotNullIfMarkedAsNullable($rule, $attribute) &&
               $this->hasNotFailedPreviousRuleIfPresenceRule($rule, $attribute);
    }
```
(see src/Illuminate/Validation/Validator.php)

The interesting part for your question is the presentOrRuleIsImplicit function.

```php

    /**
     * Determine if the field is present, or the rule implies required.
     *
     * @param  object|string  $rule
     * @param  string  $attribute
     * @param  mixed  $value
     * @return bool
     */
    protected function presentOrRuleIsImplicit($rule, $attribute, $value)
    {
        if (is_string($value) && trim($value) === '') {
            return $this->isImplicit($rule);
        }

        return $this->validatePresent($attribute, $value) ||
               $this->isImplicit($rule);
    }

    /**
     * Determine if a given rule implies the attribute is required.
     *
     * @param  object|string  $rule
     * @return bool
     */
    protected function isImplicit($rule)
    {
        return $rule instanceof ImplicitRule ||
               in_array($rule, $this->implicitRules);
    }

```

So it basically checks if the rule is in the implicitRules array:

```php

    /**
     * The validation rules that imply the field is required.
     *
     * @var array
     */
    protected $implicitRules = [
        'Accepted',
        'Filled',
        'Present',
        'Required',
        'RequiredIf',
        'RequiredUnless',
        'RequiredWith',
        'RequiredWithAll',
        'RequiredWithout',
        'RequiredWithoutAll',
    ];
```


TLDR: 
The rule integer gets ignored because it is not required. So if you want that you should add the required rule
